typings/dom-tag-component-factories.ts

```diff
@@ -0,0 +1,142 @@
+import { ThemedStyledFunction, WithOptionalTheme } from "./styled-components";
+
+export interface DomTagComponentFactory<DomTag extends keyof JSX.IntrinsicElements, Theme> {
+    <Props>(domTag: DomTag): ThemedStyledFunction<Props & JSX.IntrinsicElements[DomTag], Theme, WithOptionalTheme<Props & JSX.IntrinsicElements[DomTag], Theme>>;
+    <Props>(domTag: DomTag): ThemedStyledFunction<Props & JSX.IntrinsicElements[DomTag], Theme>;
+};
+
+
+export interface DomTagComponentFactories<Theme> extends
+  DomTagComponentFactory<"a", Theme>,
+  DomTagComponentFactory<"a", Theme>,
```

Comment:
You list `"a"` twice here.

Also, is there no way to do this with the `JSX.IntrinsicElements` list using [Mapped Types](https://www.typescriptlang.org/docs/handbook/advanced-types.html#mapped-types), or the `keyof` keyword in general? I would have thought that something like `interface DomTagComponentFactories<Theme> extends DomTagComponentFactory<keyof JSX.IntrinsicElement>`, though I'm pretty sure that that exact syntax wouldn't work...

Reply:
I tried a few things, but I couldn't find a way to limit the key to a single key in the union or to get something like mapped types to work with function overloads.

Would love if someone finds a way though :).

and will fix the double a shortly